src/TableStorage.Abstractions/TableStore.cs

```diff
@@ -608,5 +728,65 @@ public int GetRecordCount()
         }
 
         #endregion Asynchronous Methods
+
+
+        /// <summary>
+        /// Ensures the partition key is not null.
+        /// </summary>
+        /// <param name="partitionKey">The partition key.</param>
+        /// <exception cref="ArgumentNullException">partitionKey</exception>
+        private void EnsurePartitionKey(string partitionKey)
+        {
+            if (string.IsNullOrWhiteSpace(partitionKey))
+            {
+                throw new ArgumentNullException(nameof(partitionKey));
+            }
+        }
+
+        /// <summary>
+        /// Ensures the row key is not null.
+        /// </summary>
+        /// <param name="rowKey">The row key.</param>
+        /// <exception cref="ArgumentNullException">rowKey</exception>
+        private void EnsureRowKey(string rowKey)
+        {
+            if (string.IsNullOrWhiteSpace(rowKey))
+            {
+                throw new ArgumentNullException(nameof(rowKey));
+            }
+        }
+
+        /// <summary>
+        /// Builds the get by partition query.
+        /// </summary>
+        /// <param name="partitionKey">The partition key.</param>
+        /// <returns>TableQuery&lt;T&gt;.</returns>
+        private TableQuery<T> BuildGetByPartitionQuery(string partitionKey)
```

Comment:
These last set of methods could be static. It might be useful to add a region around them, something like Helpers